### PR TITLE
#61 suspense samples

### PR DESCRIPTION
**Pull request by fjcalzado**

### CODE REVIEW

File: examples/06-suspense-like/src/app.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+const Suspense: React.FC<{ tag: string, className?: string }> = ({ tag, className, children }) => {
+  const { promiseInProgress } = usePromiseTracker({ area: tag, delay: 0 });
+  return promiseInProgress ? <CircularProgress className={className} size={60}/> : <>{children}</>;
+};
+
+const AppInner: React.FC<WithStyles<typeof styles>> = ({classes}) => {
+  const [quote, setQuote] = React.useState<Quote>({ body: "", author: "" });
+  const [picture, setPicture] = React.useState();
+  const loadData = React.useCallback(() => {
```
Comment (nasdan) [inline examples/06-suspense-like/src/app.tsx:19]: Why did you use `useCallback`?

`useCallback` is to memoize the function by some params, but if you pass `[]` only memoize once.

Then, you are using it in `useEffect` but with `[]` as second parameter, that is, we are only calling `loadData` on componentDidMount.

Am I missing something?

Thank you.

File: examples/06-suspense-like/src/app.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import { WithStyles, withStyles } from "@material-ui/core/styles";
+import { trackPromise, usePromiseTracker } from "react-promise-tracker";
+import { Quote, getQuote, getPicture } from "./api";
+import { styles } from "./styles";
+
+
+const Suspense: React.FC<{ tag: string, className?: string }> = ({ tag, className, children }) => {
+  const { promiseInProgress } = usePromiseTracker({ area: tag, delay: 0 });
```
Comment (nasdan) [inline examples/06-suspense-like/src/app.tsx]: Maybe it's not necessary feed the `delay` equals 0.

**Review by nasdan — APPROVED** [review]
Really cool, thank you. Only some comments about `useCallback`

File: examples/06-suspense-like/src/app.tsx (hunk 1 above)
Comment (fjcalzado) [reply examples/06-suspense-like/src/app.tsx:19]: Cause I want to intentionally keep the same callback reference between re-renders. `loadData` does also serve as a handler for the button `onClick`.

File: examples/06-suspense-like/src/app.tsx (hunk 2 above)
Comment (fjcalzado) [reply examples/06-suspense-like/src/app.tsx]: Yes, you are right! Didn't realize it was optional.

Comment (fjcalzado) [reply examples/06-suspense-like/src/app.tsx]: Hummm .... it seems the typing file is not correct then, cause it is not optional there.

Comment (fjcalzado) [reply examples/06-suspense-like/src/app.tsx]: What a weird thing .... the .d.ts shipped with 2.0 version seems to have this:

```
interface Config {
   area: string;
   delay: number;
}
```

However, in src they are optional:

```
interface Config {
   area?: string;
   delay?: number;
}
```

Is this a bug?

Comment (fjcalzado) [reply examples/06-suspense-like/src/app.tsx]: I have removed delay param anyway, but we should check the d.ts in v2.0 package as it seems to be not in sync with the one in src. @brauliodiez are you aware of this?

Comment (brauliodiez) [reply examples/06-suspense-like/src/app.tsx]: I think in current build the are optional, I think it has been published (v 2.0.2)

```
interface Config {
   area?: string;
   delay?: number;
}
```

**Pull request merged**